app/models/dialog_field_text_box.rb

```diff
@@ -1,9 +1,9 @@
 class DialogFieldTextBox < DialogField
-  AUTOMATE_VALUE_FIELDS = %w(protected required validator_rule validator_type read_only)
+  AUTOMATE_VALUE_FIELDS = %w(data_type protected required validator_rule validator_type read_only)
 
   def value
     @value = values_from_automate if dynamic && @value.blank?
-    @value
+    data_type == "integer" ? @value.to_i : @value
```

Comment:
This forces the box always to have a value. It is 0 even if the value is not set.
When a box has a default value of `nil`, it shows up as 0 which is not desirable.